src/main/java/net/neoforged/neoforge/network/registration/NetworkRegistry.java

```diff
@@ -799,12 +800,12 @@ public boolean onVanillaNetworkConnectionEstablished(ClientConfigurationPacketLi
 
         //Because we are in vanilla land, no matter what we are not able to support any custom channels.
         sender.getConnection().channel().attr(ATTRIBUTE_PAYLOAD_SETUP).set(NetworkPayloadSetup.emptyVanilla());
-        sender.getConnection().channel().attr(ATTRIBUTE_IS_MODDED_CONNECTION).set(false);
+        sender.getConnection().channel().attr(ATTRIBUTE_CONNECTION_TYPE).set(sender.getConnectionType());
         sender.getConnection().channel().attr(ATTRIBUTE_FLOW).set(PacketFlow.CLIENTBOUND);
 
         //Negotiation failed. Disconnect the client.
         if (!configurationNegotiationResult.success()) {
-            sender.getConnection().disconnect(Component.translatableWithFallback("neoforge.network.negotiation.failure.vanilla.client.not_supported", "You are trying to connect to a server that is running NeoForge, but you are not. Please install NeoForge Version: %s to connect to this server.", NeoForgeVersion.getVersion()));
+            sender.getConnection().disconnect(Component.translatableWithFallback("neoforge.network.negotiation.failure.vanilla.server.not_supported", "You are trying to connect to a server that is not running NeoForge, but you are not. A connection could not be established.", NeoForgeVersion.getVersion()));
```

Comment:
", but you are not." is incorrect I think.

Maybe something like ", but you have mods that require it."

Reply:
The latter makes a lot more sense. This of course also applies to the line in the lang file